src/atomEffect.ts

```diff
@@ -153,7 +154,7 @@ export function atomEffect(effect: Effect): Atom<void> & { effect: Effect } {
 
       setter.recurse = (a, ...args) => {
         if (fromCleanup) {
-          if (process.env?.MODE !== 'production') {
+          if (isDev()) {
```

Comment:
This doesn't tree shake.
Unless we want a different behavior for DEV and PRD intentionally, I guess throwing always might be better.
```js
if (fromCleanup) {
  throw new Error(...);
}
```